### PR TITLE
DTRA-1469 / Kate / Refactor: ToggleSwitch and ActionSheet

### PR DESCRIPTION
**Pull request by kate-deriv**
1) Added prop `iconPosition` for specifying icon position(`left` or `right`) in the header title.
2) Fix behaviour `onFocus` for `ToggleSwitcher`: previously element always stayed focused after click/tap. By replacing `:focus `---> `:focus-visible` element stayed focused onKeyDown focusing.
3) Changed approach for ActionSheet transition: instead of timer-based animation added CSSTransition package. For that I reverted all ActionSheet changes from this [PR](https://github.com/deriv-com/quill-ui/pull/303) and added react-transition-group package.
4) Updated tests and stories.

### CODE REVIEW

File: lib/components/ActionSheet/footer/index.tsx [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import { FooterProps } from "../types";
+import { useContext } from "react";
```
Comment (kate-deriv) [inline lib/components/ActionSheet/footer/index.tsx:2]: Just reverting ActionSheet changes in this file from this [PR](https://github.com/deriv-com/quill-ui/pull/303/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519)

File: lib/components/ActionSheet/types.ts [hunk 2]
```diff
@@ -34,7 +34,6 @@ interface ActionType {
 export interface FooterProps
     extends ComponentPropsWithoutRef<"div">,
         actionSheetFooterCVA {
-    handleClose?: () => void;
```
Comment (kate-deriv) [inline lib/components/ActionSheet/types.ts:37]: Reverting ActionSheet changes in this file from this [PR](https://github.com/deriv-com/quill-ui/pull/303/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519).

File: lib/components/SelectionControl/toggle-switch/toggle-switch.scss [hunk 3]
```diff
@@ -27,7 +27,7 @@ html.dark {
     height: var(--core-size-1400);
     border-width: var(--core-borderRadius-100);
 
-    &:focus {
+    &:focus-visible {
```
Comment (kate-deriv) [inline lib/components/SelectionControl/toggle-switch/toggle-switch.scss:30]: Fix behaviour onFocus for ToggleSwitcher: previously element always stayed focused after click/tap. By replacing :focus ---> :focus-visible element stayed focused onKeyDown focusing.

File: lib/hooks/useSwipeBlock/index.ts [hunk 4]
```diff
@@ -3,15 +3,13 @@ import { useDrag } from "@use-gesture/react";
 import { useMediaQuery } from "usehooks-ts";
 
 interface SwipeBlockType {
-    expandable?: boolean;
```
Comment (kate-deriv) [inline lib/hooks/useSwipeBlock/index.ts:6]: Reverting ActionSheet changes in this file from this [PR](https://github.com/deriv-com/quill-ui/pull/303/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519).

File: package.json [hunk 5]
```diff
@@ -71,9 +71,9 @@
         "@types/node": "^20.10.7",
         "@types/react": "18.2.43",
         "@types/react-dom": "18.2.17",
-        "@types/lodash.debounce": "^4.0.7",
```
Comment (kate-deriv) [inline package.json:74]: Removed this package from this [PR](https://github.com/deriv-com/quill-ui/pull/303/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519) as it's not used anymore

**Review by maryia-deriv — APPROVED** [review]

**Pull request merged**